jsonrpcserver/errors.py

```diff
@@ -0,0 +1,27 @@
+"""Exceptions"""
+from typing import Any
+
+from .response import UNSPECIFIED
+from . import status
+
+class ApiError(RuntimeError):
```

Comment:
No, actually not. "Server errors" as defined by the standard are part of the predefined range of error codes ("The error codes from and including -32768 to -32000 are reserved for pre-defined errors."). As I understand the standard they are intended to signal errors which occurred inside the server framework code, not the server application code (so for errors internal to the jsonrpcserver lib but not errors in the code of users of the lib). So, they are IMHO perfectly suited as catch-all error if a method fails with some unexpected exception.

However, the `ApiError` is intended to signal application-defined errors with different codes, or as the standard puts it: "The remainder of the space is available for application defined errors.". That is why I named it `ApiError`: Such an exception is actually part of the public API of the interface, just like method names and parameter types.

So, we could and maybe should add an assertion in `ApiError.__init__` that the error code given to `ApiError` is NOT in the range -32768 to -32000 (so it cannot be a "Server Error"), because these codes are reserved for special use.

Reply:
You're right.